prov/opx/include/rdma/opx/fi_opx_endpoint.h

```diff
@@ -687,7 +687,11 @@ void complete_receive_operation_internal (struct fid_ep *ep,
 				ext = (struct fi_opx_context_ext *)context;
 				ext->err_entry.op_context = ext->msg.op_context;
 			} else {
-				posix_memalign((void**)&ext, 32, sizeof(struct fi_opx_context_ext));
+				if (posix_memalign((void**)&ext, 32, sizeof(struct fi_opx_context_ext))) {
+					FI_WARN(fi_opx_global.prov, FI_LOG_EP_DATA,
+						"Out of memory error.\n");
+					abort();
```

Comment:
Is there an option here that doesn't involve crashing the app?  Not that we ever expect memory allocation to fail...

Reply:
In all paths where it was possible to return an error to the caller, that's the option I chose. In these cases my options were either to just cast posix_memalign to void and pretend or this approach. I chose this because, I mean, if we've actually run out of memory the process isn't going to live much longer in any case, it's either abort() or let the OOM reaper kill us. If we manage to abort under our own power then there's at least a chance of diagnostic data.